BitwardenShared/Core/Platform/Models/Domain/ServerConfig.swift

```diff
@@ -39,6 +39,28 @@ struct ServerConfig: Equatable, Codable {
         server = responseModel.server.map(ThirdPartyServerConfig.init)
         version = responseModel.version
     }
+
+    func isServerVersionAfter() -> Bool {
+        let cleanServerVersion = version.split(separator: "-").first ?? ""
+        let cleanMinServerVersion = Constants.CipherKeyEncryptionMinServerVersion.split(separator: "-").first ?? ""
+
+        let serverVersion = cleanServerVersion.split(separator: ".").map { Int($0) ?? 0 }
+        let minServerVersion = cleanMinServerVersion.split(separator: ".").map { Int($0) ?? 0 }
+
+        if serverVersion.isEmpty || minServerVersion.isEmpty {
+            return false
+        }
+
+        for (serverV, minServerV) in zip(serverVersion, minServerVersion) {
+            if serverV < minServerV {
+                return false
+            } else if serverV > minServerV {
+                return true
+            }
+        }
+
+        return serverVersion.count >= minServerVersion.count
+    }
```

Comment:
🎨 Even though this is mostly good, I'd make some improvements:
- I'd create a `Version` struct that would take care of passing a `String`, cleaning it and splitting the components accordingly to use (defaulting the ones that don't have value because of wrong formatting or throwing an error depending on how you'd like to tackle it).
- In such struct I'd add the separators (`-` and `.`) as constants so they're not hardcoded on each use.
- I'd conform to `Comparable` and do the comparison logic there.
- So then you could simply do `Version("2024.3.1") > Version("2024.2.3")`, therefore it's easier to read and use.

🤔 Aside from the above, not sure why you're doing `return serverVersion.count >= minServerVersion.count`, here you're checking if both versions have the same number of components whereas IMO you should check that before zipping the versions and if they don't have the same number of components then you could fill the gaps with `-1` or just fail depending on how you'd like to tackle it. Therefore, at the end if it goes out of the `for` it means both versions are equal and you can return that they are.

Reply:
🎨 I would potentially call it `ServerVersion` and make sure it's also `Codable` but otherwise I agree with Fede that turning a version into its own object (that's a lot easier to write tests around vis-à-vis parsing server versions and handling errors) would be ideal. The `serverVersion` variable should then be of that type, which might require a migration and/or you expose a `Version` object that's a computed property off of the `serverVersion`.

Then at the calling site you can just say `config.serverVersion > Version("2024.2.3")` or whatever and it's taken care of; it'll also make writing tests in other places easier, I feel.